featurebyte/core/accessor/target_string.py

```diff
@@ -0,0 +1,278 @@
+
+from typing import TYPE_CHECKING, Optional, TypeVar
+
+from featurebyte.common.doc_util import FBAutoDoc
+from featurebyte.core.accessor.string import StringAccessor
+from featurebyte.query_graph.node.string import Side
+
+if TYPE_CHECKING:
+    from featurebyte.api.target import Target
+else:
+    Target = TypeVar("Target")
+
+
+class TargetStrAccessorMixin:
+    """
+    TargetStrAccessorMixin class
+    """
+
+    @property
+    def str(self: Target) -> StringAccessor:  # type: ignore # pylint: disable=invalid-name
+        """
+        str accessor object
+
+        Returns
+        -------
+        StringAccessor
+        """
+        return TargetStringAccessor(self)
+
+
+class TargetStringAccessor(StringAccessor):
+    """
+    TargetStringAccessor class used to manipulate string type Target objects
+    """
+
+    # documentation metadata
+    __fbautodoc__ = FBAutoDoc()
+
+    def len(self) -> Target:
+        """
+        Computes length of each string element.
+
+        Returns
+        -------
+        Target
+            A new Target object.
+
+        Examples
+        --------
+        Compute the length of each string element in the ProductGroupLookup target:
+
+        >>> target = catalog.get_target("ProductGroupLookup")  # doctest: +SKIP
+        >>> target["ProductGroupLength"] = target.str.len()  # doctest: +SKIP
+        """
+        return super().len()  # type: ignore[return-value]
+
+    def lower(self) -> Target:
+        """
+        Converts each string element to lower case.
+
+        Returns
+        -------
+        Target
+            A new Target object.
+
+        Examples
+        --------
+        Convert the ProductGroupLookup target to lower case:
+
+        >>> target = catalog.get_target("ProductGroupLookup")  # doctest: +SKIP
+        >>> target["ProductGroupLower"] = target.str.lower()  # doctest: +SKIP
+        """
+        return super().lower()  # type: ignore[return-value]
+
+    def upper(self) -> Target:
+        """
+        Converts each string element to upper case.
+
+        Returns
+        -------
+        Target
+            A new Target object.
+
+        Examples
+        --------
+        Convert the ProductGroupLookup target to upper case:
+
+        >>> target = catalog.get_target("ProductGroupLookup")  # doctest: +SKIP
+        >>> target["ProductGroupUpper"] = target.str.upper()  # doctest: +SKIP
+        """
+        return super().upper()  # type: ignore[return-value]
+
+    def strip(self, to_strip: Optional[str] = None) -> Target:
+        """
+        Removes leading and trailing characters (whitespaces by default) from each string element.
+
+        Parameters
+        ----------
+        to_strip: Optional[str]
+            Characters to remove, whitespace is used by default.
+
+        Returns
+        -------
+        Target
+            A new Target object.
+
+        Examples
+        --------
+        Remove leading and trailing "M" characters from the Title column:
```

Comment:
The term `column` or `Series` are sometimes still referenced in these docstrings. Is that common in the existing docstrings?

Reply:
yea, some of the feature docstrings reference that right now. let me clean it up.